### PR TITLE
Remove extra parameter from deleteDataSourceSpecificInterval endpoint and correct exception message for invalid interval

### LINKED ISSUE #2532: DELETE datasource/{dataSourceName}/intervals/{interval} does not need "kill" argument
PR https://github.com/druid-io/druid/pull/2276 added a new endpoint for deleting segments:
   /druid/coordinator/v1/datasources/{dataSourceName}/intervals/{interval}?kill=true
   
If kill is set to false, it returns an error.  Parag has suggested that the kill argument is not needed, that we should just let the endpoint be
   DELETE
   /druid/coordinator/v1/datasources/{dataSourceName}/intervals/{interval}

### PR DESCRIPTION
**Pull request by pjain1**
Fixes #2532 
- Remove extra kill parameter
- Correct exception message for invalid interval

### CODE REVIEW

**Comment by drcrallen** [comment]
@pjain1 can you move the `Fixes #2532` into the PR master comment and make the title of this PR more descriptive?

**Comment by pjain1** [comment]
@drcrallen done

File: server/src/main/java/io/druid/server/http/DatasourcesResource.java [hunk 1]
```diff
@@ -180,15 +179,28 @@ public Response deleteDataSource(
...
         indexingServiceClient.killSegments(dataSourceName, new Interval(interval));
       }
-      catch (Exception e) {
+      catch (IllegalArgumentException e) {
+        return Response.serverError().entity(
+            ImmutableMap.of(
+                "error",
+                "Exception occurred. Probably the interval is invalid",
+                "Error-Message",
+                e
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: usually this is `"message":e.getMessage()`

File: server/src/main/java/io/druid/server/http/DatasourcesResource.java [hunk 2]
```diff
@@ -180,15 +179,28 @@ public Response deleteDataSource(
...
+            )
+        )
+                       .build();
+      } catch (Exception e) {
         return Response.serverError().entity(
             ImmutableMap.of(
                 "error",
-                "Exception occurred. Are you sure you have an indexing service?"
+                "Exception occurred. Are you sure you have an indexing service?",
+                "Error-Message",
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: usually this is `"message":e.getMessage()`

File: server/src/main/java/io/druid/server/http/DatasourcesResource.java [hunk 3]
```diff
@@ -180,15 +179,28 @@ public Response deleteDataSource(
...
       return Response.ok(ImmutableMap.of("error", "no indexing service found")).build();
     }
+
     if (kill != null && Boolean.valueOf(kill)) {
       try {
         indexingServiceClient.killSegments(dataSourceName, new Interval(interval));
       }
-      catch (Exception e) {
+      catch (IllegalArgumentException e) {
+        return Response.serverError().entity(
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: This is not server error, this is bad request

File: server/src/main/java/io/druid/server/http/DatasourcesResource.java (hunk 1 above)
Comment (pjain1) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: yes...but I wanted to have `java.lang.IllegalArgumentException` to be part of the `Error-Message`. If I use `e.getMessage()` it does not include the Exception class

File: server/src/main/java/io/druid/server/http/DatasourcesResource.java (hunk 3 above)
Comment (pjain1) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: done. Thanks

File: server/src/main/java/io/druid/server/http/DatasourcesResource.java (hunk 2 above)
Comment (pjain1) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: yes but I wanted to have `Exception` class name to be part of the `Error-Message`. If I use `e.getMessage()` it does not include the Exception class

Comment (gianm) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: Does this get toString'd automatically? If not I think it should be `e.toString()`

Comment (pjain1) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: yes, it does that automatically

**Comment by rasahner** [comment]
It might be useful to put a comment on the deprecated function deleteDataSource to say that
it's only the part for kill != null that's deprecated.  If the whole function were removed, there'd be no support for the endpoint "DELETE /druid/coordinator/v1/datasources/{dataSourceName}" (with no parameters).

**Comment by pjain1** [comment]
@rasahner done, please verify

**Comment by rasahner** [comment]
that's fine, thanks very much

**Comment by fjy** [comment]
:+1:

**Comment by fjy** [comment]
@rasahner can you +1 this if you think it is ready?

**Comment by rasahner** [comment]
+1

**Pull request merged**